nixos/modules/tasks/lvm.nix

```diff
@@ -1,17 +1,61 @@
 { config, lib, pkgs, ... }:
 
 with lib;
-
-{
-
-  ###### implementation
-
-  config = mkIf (!config.boot.isContainer) {
-
-    environment.systemPackages = [ pkgs.lvm2 ];
-
-    services.udev.packages = [ pkgs.lvm2 ];
-
+let
+  cfg = config.services.lvm;
+in {
+  options.services.lvm = {
+    dmeventd.enable = mkEnableOption "the LVM dmevent daemon";
+    boot.thin.enable = mkEnableOption "support for booting from ThinLVs";
   };
 
+  config = let
+    pkg = if cfg.dmeventd.enable then pkgs.lvm2_dmeventd else pkgs.lvm2;
```

Comment:
Maybe this should be exposed as an (internal) option, in case someone has their own special lvm2 package with more units or something. Sure they could also just overlay the module, but that would mean a whole bunch of rebuilds.

Reply:
Good idea, exposed as an internal option.